drivers/gles2/shader_compiler_gles2.cpp

```diff
@@ -229,17 +229,18 @@ void ShaderCompilerGLES2::_dump_function_deps(const SL::ShaderNode *p_node, cons
 
 	ERR_FAIL_COND(fidx == -1);
 
-	for (Set<StringName>::Element *E = p_node->functions[fidx].uses_function.front(); E; E = E->next()) {
-		if (r_added.has(E->get())) {
+	for (int j = 0; j < p_node->functions[fidx].uses_function.size(); j++) {
```

Comment:
I'd rename the inner index to avoid confusion, I've never really seen an outer index `j` and an inner `i`, or name it more descriptively like `function_i`

Reply:
I think i'll name it more descriptively to avoid unnecessary changes (there are two inner loops).